lib/dice/src/handoff.rs

```diff
@@ -0,0 +1,73 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+use crate::{AliasCert, AliasOkm, DeviceIdCert};
+use hubpack::SerializedSize;
+use lpc55_pac::syscon::RegisterBlock;
+use serde::{Deserialize, Serialize};
+
+// This memory is the USB peripheral SRAM that's 0x4000 bytes long. Changes
+// to this address must be coordinated with the [dice_*] tables in
+// chips/lpc55/chips.toml
+const MEM_START: usize = 0x4010_0000;
+const ALIAS_START: usize = MEM_START;
+
+pub struct Handoff<'a>(&'a RegisterBlock);
+
+impl<'a> Handoff<'a> {
+    pub fn turn_on(syscon: &'a RegisterBlock) -> Self {
+        // handoff through USB SRAM requires we power it on
+        syscon.ahbclkctrl2.modify(|_, w| w.usb1_ram().enable());
+        syscon
+            .presetctrl2
+            .modify(|_, w| w.usb1_ram_rst().released());
+
+        Self(syscon)
+    }
+
+    pub fn turn_off(self) {
+        self.0
+            .presetctrl2
+            .modify(|_, w| w.usb1_ram_rst().asserted());
+        self.0.ahbclkctrl2.modify(|_, w| w.usb1_ram().disable());
+    }
+
+    pub fn alias(&self, alias_handoff: &AliasHandoff) -> usize {
+        let dst_ptr = ALIAS_START as *mut [u8; AliasHandoff::MAX_SIZE];
+        // SAFETY: Dereferencing this raw pointer is necessary to write to the
+        // memory region used to handoff DICE artifacts to Hubris tasks. This
+        // pointer will always references a valid memory region.
```

Comment:
Explain why you know it will always be valid

Reply:
This was a pretty bad safety comment. Apologies for that. I've updated both SAFETY comments in this file to include an enumeration of the preconditions that must be met for the call to be safe.